scripts/SREP/infra-node-resize/script.sh

```diff
@@ -0,0 +1,90 @@
+    oc -n "$namespace" get machinepool "$INFRA_MACHINEPOOL" -o json | jq -r ".spec.platform"
+}
+
+# update the yaml file
+update_cluster_infra_yaml() {
+    oc -n "$namespace" get machinepool "$INFRA_MACHINEPOOL" -o yaml > /tmp/cluster-infra.yaml
+    yq -i 'del(.metadata.creationTimestamp) | del(.metadata.finalizer) | del(.metadata.resourceVersion) | del(.metadata.generation) | del(.metadata.selfLink) | del(.metadata.uid) | del(.status)' /tmp/cluster-infra.yaml
+    cat /tmp/cluster-infra.yaml 
+    CLOUD=$(oc -n "$namespace" get machinepool "$INFRA_MACHINEPOOL" -o json | jq -r '.spec.platform | keys_unsorted[0]')
+    yq -i ".metadata.name |= . + "2" | .spec.name |= . + "2" | .spec.platform."$CLOUD".type |= env(INSTANCE_SIZE)" /tmp/cluster-infra.yaml 
+    cat /tmp/cluster-infra.yaml
+}
+
+# create the temporary machinepool
+create_temp_machinepool() {
+    oc -n "$namespace" create -f /tmp/cluster-infra.yaml
+    while [ -z "$(oc get machinepool -n "$namespace" "$INFRA_MACHINEPOOL" -o=jsonpath='{.status.machineSets[0].errorMessage}')" ];
+    do
+        echo 'Machines are still not correctly provisioned'
+        sleep 10
+    done
+}
+
+# delete the original machinepool
+delete_original_machinepool() {
+    oc scale --replicas 0 machinepool "$INFRA_MACHINEPOOL" -n "$namespace"
+    oc -n "$namespace" get machinepool
+    while [ -z "$(oc get machinepool -n "$namespace" "$INFRA_MACHINEPOOL" -o=jsonpath='{.status.machineSets[0].errorMessage}')" ];
+    do
+        echo 'Machines are still not properly deleted'
+        sleep 10
+    done
+    oc -n "$namespace" delete machinepool "$INFRA_MACHINEPOOL"
+    echo 'Original machinepools which are not resized are deleted'
+}
+
+# update yaml file for new cluster 
+update_new_cluster_infra_yaml() {
+    yq -i '.metadata.name |= sub("2", "") | .spec.name |= sub("2", "")' /tmp/cluster-infra.yaml 
+}
+
+# create new machinepool
+create_new_machinepool() {
+    oc -n "$namespace" create -f /tmp/cluster-infra.yaml
+    while [ -z "$(oc get machinepool -n "$namespace" "$INFRA_MACHINEPOOL" -o=jsonpath='{.status.machineSets[0].errorMessage}')" ];
+    do
+        echo 'Machines are still not correctly provisioned'
+        sleep 10
+    done
+}
+
+# delete temporary machinepool 
+delete_temp_machinepool() {
+    oc scale --replicas 0 machinepool "${INFRA_MACHINEPOOL}"2 -n "$namespace"
+    oc -n "$namespace" get machinepool
+    while [ -z "$(oc get machinepool -n "$namespace" "$INFRA_MACHINEPOOL" -o=jsonpath='{.status.machineSets[0].errorMessage}')" ];
+    do
+        echo 'Machines are still not properly deleted'
+        sleep 10
+    done
+    oc -n "$namespace" delete machinepool "${INFRA_MACHINEPOOL}"2 
+    echo 'Temporary machinepools are deleted'
+}
+
+main() {
+  get_current_infra_size
+  update_cluster_infra_yaml
+  create_temp_machinepool
+  delete_original_machinepool
+  update_new_cluster_infra_yaml
+  create_new_machinepool
+  delete_temp_machinepool
+  echo "Succeed."
```

Comment:
An echo statement on the entry of each step will be helpful to let the user know which step the script currently is. Also, would be helpful in case of script failure